presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -113,10 +113,11 @@ public synchronized void removeQuery(QueryId queryId)
     }
 
     @VisibleForTesting
-    public synchronized void collectDynamicFilters()
+    public void collectDynamicFilters()
```

Comment:
it would be good to also reduce synchronization in `registerQuery`, so that stream collection happens before the lock is taken.